src/components/data_submission/ds_common.js

```diff
@@ -0,0 +1,341 @@
+    formInfo
+  } = config;
+
+  return div({ style: { ...styles.flexRow, justifyContent: 'unset', alignItems: 'center' } }, [
+    label({ className: 'switch', htmlFor: `cb_${id}_${toggleText}` }, [
+      input({
+        type: 'checkbox',
+        id: `cb_${id}_${toggleText}`,
+        checked: get(id, formInfo) === undefined ? defaultValue : get(id, formInfo),
+        className: 'checkbox-inline',
+        onChange: (event) => onFormInputChange(config, event.target.checked),
+        disabled
+      }),
+      div({className: 'slider round'}),
+    ]),
+    div({
+      style: {
+        marginLeft: 15,
+        fontStyle: 'italic'
+      }
+    }, [toggleText])
+  ]);
+};
+
+//---------------------------------------------
+// Main Components
+//---------------------------------------------
+/*
+* Config options:
+* id, title, description
+* type (ENUM: 'text', 'multitext', 'select', 'sliding-checkbox')
+*  * type == select
+*    * selectOptions: [{key: string, displayText: string}]
+*    * searchPlaceholder
+*  * type == 'checkbox'
+*    * toggleText
+*    * checkboxType
+* required, disabled
+* placeholder, defaultValue,
+* style (for the formControl component)
+* inputStyle (for the input element)
+* onChange,
+* errors, setErrors,
+* formInfo, setFormInfo
+*/
+export const FormField = (config) => {
+  const {
+    id, title, hideTitle, description,
+    required, style, errors
+  } = config;
+
+  return div({
+    key: `formControl_${id}`,
+    style,
+    className: `formField-container formField-${id}`
+  }, [
+    title && !hideTitle && label({
+      id: `lbl_${id}`,
+      className: `control-label ${errors[id] ? 'errored' : ''}`,
+      htmlFor: `${id}`
+    }, [
+      title,
+      required && '*'
+    ]),
+    description && div({ style: { marginBottom: 15 } }, [description]),
+    formInput(config)
+  ]);
+};
+
+/*
+* Config options:
+* id
+* formFields: array[FormField Configs]
+* onChange, errors, setErrors, formInfo, setFormInfo
+*/
+export const FormTable = (config) => {
+  const {
+    id, formFields,
+    onChange, errors, setErrors, formInfo, setFormInfo
+  } = config;
+
+  return div({ id, className: `formTable` }, [
+    // generate columns
+    div({ className: 'formTable-row' }, formFields.map(x => {
+      return label({ className: 'control-label', id: `${id}-${x.title}` }, [x.title]);
+    })),
+    // generate form rows
+    get(id, formInfo).map((formRow, i) => {
+      return div({ className: 'formTable-row', key: `formtable-${id}-${i}` }, formFields.map(formCol => {
+        return FormField({
+          ...formCol,
+          id: `${id}-${i}-${formCol.id}`,
+          path: `${id}.${i}.${formCol.id}`,
+          hideTitle: true, ariaDescribedby: `${id}-${formCol.title}`,
+          onChange, errors, setErrors, formInfo, setFormInfo
+        });
+      }));
+    }),
+    // add new button
+    div({ style: { display: 'flex', width: '100%', justifyContent: 'flex-end' } }, [
```

Comment:
what is the add new button for? maybe this should exist outside of `FormTable` since the component could be used for more things.